File: src/main/java/com/dh/ondot/schedule/api/response/HomeScheduleListResponse.java
package com.dh.ondot.schedule.api.response;

import com.dh.ondot.core.util.TimeUtils;
import com.dh.ondot.schedule.application.dto.HomeScheduleListItem;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.List;

public record HomeScheduleListResponse(
        Long earliestAlarmId,
        LocalDateTime earliestAlarmAt,
        List<HomeScheduleListItem> scheduleList,
        boolean hasNext
) {
    public static HomeScheduleListResponse of(
            Long earliestAlarmId,
            Instant earliestAlarmAt,
            List<HomeScheduleListItem> list,
            boolean hasNext
    ) {
        return new HomeScheduleListResponse(
                earliestAlarmId,
                TimeUtils.toSeoulDateTime(earliestAlarmAt),
                list,
                hasNext
        );
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/api/response/HomeScheduleListResponse.kt
package com.dh.ondot.schedule.api.response

import com.dh.ondot.core.util.TimeUtils
import com.dh.ondot.schedule.application.dto.HomeScheduleListItem
import java.time.Instant
import java.time.LocalDateTime

data class HomeScheduleListResponse(
    val earliestAlarmId: Long?,
    val earliestAlarmAt: LocalDateTime?,
    val scheduleList: List<HomeScheduleListItem>,
    val hasNext: Boolean,
) {
    companion object {
        @JvmStatic
        fun of(
            earliestAlarmId: Long?,
            earliestAlarmAt: Instant?,
            list: List<HomeScheduleListItem>,
            hasNext: Boolean,
        ): HomeScheduleListResponse {
            return HomeScheduleListResponse(
                earliestAlarmId = earliestAlarmId,
                earliestAlarmAt = TimeUtils.toSeoulDateTime(earliestAlarmAt),
                scheduleList = list,
                hasNext = hasNext,
            )
        }
    }
}
